analytic_engine/src/table/mod.rs

```diff
@@ -112,19 +359,21 @@ impl Table for TableImpl {
     }
 
     async fn write(&self, request: WriteRequest) -> Result<usize> {
+        if self.instance.max_rows_in_write_queue > 0 {
```

Comment:
Add function `enable_pending_write()` to encapsulate this.

Reply:
Add a method `should_queue_write_request` to do this check.